contents/swift/transition-to-swift.md

```diff
@@ -0,0 +1,340 @@
+
+## [Strings and Characters](https://developer.apple.com/library/content/documentation/Swift/Conceptual/Swift_Programming_Language/StringsAndCharacters.html#//apple_ref/doc/uid/TP40014097-CH7-ID285)
+
+### Things to note:
+
+- The escaped special characters \0 (null character), \\ (backslash), \t (horizontal tab), \n (line feed), \r (carriage return), \" (double quote) and \' (single quote).
+
+- You can use + to concatenate Strings and use indexes directly on Strings.
+
+  ```
+  var str1 = "This is the first part "
+  var str2 = "of this sentence.\n"
+  var str3 = str1 + str2 //"This is the first part of this sentence.\n"
+  var tenthChar = str3[9] //'h'
+  ```
+
+## [Collection Types](https://developer.apple.com/library/content/documentation/Swift/Conceptual/Swift_Programming_Language/CollectionTypes.html)
+
+### Things to note:
+
+- A dictionary is similar to HashMap in other languages.
+
+
+### Other references:
+
+- Lesser-known Foundation Data Structures: https://www.raywenderlich.com/123100/collection-data-structures-swift-2
+
+  If a dictionary, array or set won’t do the job, it’s worth checking if one of these will work before you create something from scratch.
+
+
+## [Control Flow](https://developer.apple.com/library/content/documentation/Swift/Conceptual/Swift_Programming_Language/ControlFlow.html)
+
+### Things to note:
+
+- For loop example:<br>
+  ```
+  for i in 0...100 {
+    print(i)
+  }
+  ```
+
+- Repeat-While is the equivalent of Do-While in other languages.
+
+- Switch does not require an explicit break for each case as switch statements in Swift do not fall through the bottom of each case and into the next one by default.
+
+  ```
+  let integerToDescribe = 5
+  var description = "The number \(integerToDescribe) is"
+  switch integerToDescribe {
+  case 2, 3, 5, 7, 11, 13, 17, 19:
+      description += " a prime number, and also"
+      fallthrough
+  default:
+      description += " an integer."
+  }
+  print(description)
+  // Prints "The number 5 is a prime number, and also an integer."
+  ```
+
+  There are many other subtle differences in switch-case for Swift, such as interval matching, testing multiple values in one statement using tuples, able to bind values in switch-cases to constants and variables, and using `where` to check for additional conditions.
+
+- The term `guard` is similar to an `if` statement, but it always has an `else` clause. It is generally used in the case of error handling, to guard against invalid(`nil`) variables that are used in the later part of the code. When the `guard` statement test fails, the `else` clause is used to exit the current method or loop, so that the invalid variable is not accessed. This will be covered more in Error-Handling.
```

Comment:
I don't really understand this, perhaps show a code example? Or you should just reference Error-Handling because it feels out of place.